### PR TITLE
Expose additional task definition fields

### PR DESCRIPTION
**Pull request by FedericoNegri**

### CODE REVIEW

**Comment by FedericoNegri** [comment]
@PipKat can you please review doc changes in the `src/ansys/hps/client/jms/schema` folder? (those in `src/ansys/hps/client/jms/resources` are autogenerated accordingly). Thanks

File: .github/actions/hps_services/action.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+name: 'Local HPS Deployment'
+
+description: 'Docker Compose based local HPS deployment'
```
Comment (PipKat) [inline .github/actions/hps_services/action.yml:3]:
```suggestion
description: 'Docker Compose-based local HPS deployment'
```

File: src/ansys/hps/client/jms/schema/task_definition.py [hunk 2]
```diff
@@ -75,6 +101,14 @@ class Meta(BaseSchema.Meta):
         allow_none=True,
         metadata={"description": "Whether to enable distributed parallel processing."},
     )
+    compute_resource_set_id = fields.String(
+        allow_none=True,
+        metadata={"description": "ID of the compute resource set this task should run on."},
```
Comment (PipKat) [inline src/ansys/hps/client/jms/schema/task_definition.py:106]:
```suggestion
        metadata={"description": "ID of the compute resource set that this task should run on."},
```

File: src/ansys/hps/client/jms/schema/task_definition.py [hunk 3]
```diff
@@ -75,6 +101,14 @@ class Meta(BaseSchema.Meta):
...
+    )
+    evaluator_id = fields.String(
+        allow_none=True,
+        metadata={"description": "ID of the evaluator this task should run on."},
```
Comment (PipKat) [inline src/ansys/hps/client/jms/schema/task_definition.py:110]:
```suggestion
        metadata={"description": "ID of the evaluator that this task should run on."},
```

**Review by PipKat — APPROVED** [review]
Just a few suggestions to improve readability.

File: src/ansys/hps/client/jms/resource/task_definition.py [hunk 4]
```diff
@@ -54,6 +55,10 @@ class HpcResources(Object):
         Name of the job scheduler queue.
     use_local_scratch : bool, optional
         Whether to use node local storage.
+    native_submit_options : str, optional
+        Additional command line options to pass directly to the scheduler.
+    custom_orchestration_options : dict[str, Union[int, float, str, bool]], optional
+        A dictionary of custom orchestration options.
```
Comment (PipKat) [inline src/ansys/hps/client/jms/resource/task_definition.py:61]:
```suggestion
        Dictionary of custom orchestration options.
```
We omit the leading article (A, An, The).

File: src/ansys/hps/client/jms/resource/task_definition.py [hunk 5]
```diff
@@ -156,6 +173,31 @@ def __init__(self, name: str = missing, version: str = missing, **kwargs):
 SoftwareSchema.Meta.object_class = Software
 
 
+class WorkerContext(Object):
+    """Provides the worker context resource.
+
+    Parameters
+    ----------
+    max_runtime : int, optional
```
Comment (PipKat) [inline src/ansys/hps/client/jms/resource/task_definition.py:181]: Usually, when "optional" is stated, we have a line immediately after the brief description like this: "The default is X." Some libraries specify the default like this immediately after the data type, but I don't know if yours supports this.

max_runtime : int, default: 30

File: src/ansys/hps/client/jms/resource/task_definition.py [hunk 6]
```diff
@@ -156,6 +173,31 @@ def __init__(self, name: str = missing, version: str = missing, **kwargs):
...
+        Maximum run time (in seconds) for an ephemeral worker.
+    max_num_parallel_tasks : int, optional
+        Maximum number of tasks a worker can run in parallel.
```
Comment (PipKat) [inline src/ansys/hps/client/jms/resource/task_definition.py:184]:
```suggestion
        Maximum number of tasks that a worker can run in parallel.
```

File: src/ansys/hps/client/jms/schema/task_definition.py [hunk 7]
```diff
@@ -58,6 +58,32 @@ class Meta:
...
+        allow_none=True,
+        metadata={
+            "description": "Additional command line options to pass directly to the scheduler."
+        },
+    )
+    custom_orchestration_options = fields.Dict(
+        allow_none=True,
+        keys=fields.Str(),
+        values=RestrictedValue(),
+        metadata={"description": "A dictionary of custom orchestration options."},
```
Comment (PipKat) [inline src/ansys/hps/client/jms/schema/task_definition.py:71]:
```suggestion
        metadata={"description": "Dictionary of custom orchestration options."},
```

File: src/ansys/hps/client/jms/schema/task_definition.py [hunk 8]
```diff
@@ -58,6 +58,32 @@ class Meta:
...
+    class Meta(BaseSchema.Meta):
+        pass
+
+    max_runtime = fields.Int(
+        allow_none=True,
+        metadata={"description": "Maximum run time (in seconds) for an ephemeral worker."},
+    )
+    max_num_parallel_tasks = fields.Int(
+        allow_none=True,
+        metadata={"description": "Maximum number of tasks a worker can run in parallel."},
```
Comment (PipKat) [inline src/ansys/hps/client/jms/schema/task_definition.py:85]:
```suggestion
        metadata={"description": "Maximum number of tasks that a worker can run in parallel."},
```

File: src/ansys/hps/client/jms/resource/task_definition.py (hunk 5 above)
Comment (FedericoNegri) [reply src/ansys/hps/client/jms/resource/task_definition.py:181]: Doesn't apply here. Optional fields are defaulted server-side, most often to null.

**Review by ojkoenig — APPROVED** [review]

**Pull request merged**